src/cargo/core/compiler/build_config.rs

```diff
@@ -68,7 +68,11 @@ impl BuildConfig {
                  its environment, ignoring the `-j` parameter",
             )?;
         }
-        let jobs = jobs.or(cfg.jobs).unwrap_or(::num_cpus::get() as u32);
+        let jobs = match jobs.or(cfg.jobs) {
+            None => ::num_cpus::get() as u32,
+            Some(j) if j != 0 => (::num_cpus::get() as u32).max(j),
```

Comment:
Why does it call `num_cpus` with `max` here?  This effectively prevents `-j` from working.

Reply:
That's a logic error on my part. Would the correct behavior be accepting the `-j` parameter regardless of its value as long as it's greater than 0?

Reply:
Yea, I think so.

Reply:
Ok hopefully that's taken care of by this latest commit